src/azure-cli-core/azure/cli/core/_profile.py

```diff
@@ -182,16 +182,22 @@ def _set_subscriptions(self, new_subscriptions):
                 s[_IS_DEFAULT_SUBSCRIPTION] = False
 
             if not new_active_one:
-                new_active_one = new_subscriptions[0]
-            new_active_one[_IS_DEFAULT_SUBSCRIPTION] = True
-            default_sub_id = new_active_one[_SUBSCRIPTION_ID]
+                new_active_one = Profile._pick_working_subscription(new_subscriptions)
         else:
-            new_subscriptions[0][_IS_DEFAULT_SUBSCRIPTION] = True
-            default_sub_id = new_subscriptions[0][_SUBSCRIPTION_ID]
+            new_active_one = Profile._pick_working_subscription(new_subscriptions)
+
+        new_active_one[_IS_DEFAULT_SUBSCRIPTION] = True
+        default_sub_id = new_active_one[_SUBSCRIPTION_ID]
 
         set_cloud_subscription(active_cloud.name, default_sub_id)
         self._storage[_SUBSCRIPTIONS] = subscriptions
 
+    @staticmethod
+    def _pick_working_subscription(subscriptions):
+        from azure.mgmt.resource.subscriptions.models import SubscriptionState
+        s = next((x for x in subscriptions if x['state'] == SubscriptionState.enabled.value), None)
+        return s or subscriptions[0]
```

Comment:
The caller of `_set_subscriptions` will ensure the subscriptions will not be empty, so we are fine. Thanks for asking.